api/envoy/extensions/filters/http/custom_response/v3/custom_response.proto

```diff
@@ -0,0 +1,86 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.custom_response.v3;
+
+import "envoy/config/core/v3/base.proto";
+import "envoy/config/core/v3/substitution_format_string.proto";
+
+import "google/protobuf/wrappers.proto";
+
+import "xds/annotations/v3/status.proto";
+import "xds/type/matcher/v3/matcher.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.custom_response.v3";
+option java_outer_classname = "CustomResponseProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/custom_response/v3;custom_responsev3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+option (xds.annotations.v3.file_status).work_in_progress = true;
+
+// [#protodoc-title: Custom Response Filter]
+// [#extension: envoy.filters.http.custom_response]
+
+// The Custom Response Filter allows for replacing upstream responses.
+
+// The filter configuration is a collection of custom response
+// policies.
+message CustomResponse {
+  // Definition of the custom response to be provided to the downstream.
+  // [#next-free-field: 6]
+  message Response {
+    message RemoteDataSource {
+      // The Http URI to fetch the custom response from.
```

Comment:
I think this comment is misleading, since Envoy does not fetch the response, but rather than redirects the client to this URL and handles redirect internally. There are some significant differences:

1. Extensions for the redirected URL will run
2. request header, included these added by filter chain during request processing will be sent to the redirected origin.
3. There is a comment that it does not work for internally generated responses (i.e. sendLocalReply). Is this restriction ok? I think this feature was designed for cases where control plane had failed to provide any endpoints for clusters.

Reply:
Sorry about that, this was true when the implementation used the Http async client. I will update it.

Regarding 3, as implemented, this feature is incompatible with sendLocalReply for `remote` responses (and not `local`), if sendLocalReply is sent after the redirection as well. 

If the control plane fails to provide clusters for the original request, the redirection should succeed if the cluster is found for the redirected route. 

Further if a custom response redirection is triggered by a sendLocalReply for a route-specific config (used for authentication for eg.) which is not exercised for the redirected route, the redirection will succeed.